### ISSUE TITLE
[Suggestion] Automatic update before exit (instead of after start)

### ISSUE THREAD

**Issue by Rookzor** [Epix37/Hearthstone-Deck-Tracker#2151]
Hello, its great how often new versions are released, but sometimes I just want to play the game right away, but still want to have the Decktracker updated for next time. This could be achieved if the updated started automatically before exit. Would work great especially with "Close with Hearthstone" option enabled.

**Comment by Cu8eR** [comment]
I dont agree - update is sometimes neccesary to follow changes in HS client to a proper logging.

It is 10 second job, not more

**Comment by Rookzor** [comment]
Well there are updates much more often than Hearthstone client, like couple times per week. And sometimes (granted not too often) the update get stuck for some time for whatever reason, might be connection, but the point stands.

**Comment by Epix37** [comment]
@Cu8eR has a valid point. Plus I perfer not downloading things without the user knowing.

> couple times per week

That's a bit of an exaggeration:
* `v0.13.6/7`: Dec 18
* `v0.13.8/9`: Jan 29
* `v0.13.10/11`: Feb 17
* `v0.13.12`: Feb 23

**Comment by Rookzor** [comment]
I'm not saying updating is a bad thing! I just want to do it in the downtime, which is, for me at least, after I close Hearthstone. 

If the popup still appeared at that point, that would be perfect, so it would just be swapping the moment when it appeared. Or it could even have the confirmation screen asking [Update / Not now / Update at exit]. I didn't actually mean it would update the client without asking, that was poor wording on my part.

My issue is that I usually click "not now" option at start, then some time later when I close the game I remember I wanted to update the tracker, and I have to restart the it again. I haven't found a "Update now" button in the menu that would force the update after I already clicked "not now" without the need to restart the client, but maybe its there and I'm just blind, can't rule that out :)

About the versions, I guess it just felt like it was more often because I always forgot to update it after I closed the game.
